includes/multi-currency/class-frontend-prices.php

```diff
@@ -65,15 +74,20 @@ public function __construct( Multi_Currency $multi_currency ) {
 	 * Returns the price for a product.
 	 *
 	 * @param mixed $price The product's price.
+	 * @param mixed $product WC_Product or null.
 	 *
 	 * @return mixed The converted product's price.
 	 */
-	public function get_product_price( $price ) {
+	public function get_product_price( $price, $product = null ) {
 		if ( ! $price ) {
 			return $price;
 		}
 
-		return $this->multi_currency->get_price( $price, 'product' );
+		if ( $this->compatibility->should_convert_product_price( $product ) ) {
+			return $this->multi_currency->get_price( $price, 'product' );
+		}
```

Comment:
I inverted it in https://github.com/Automattic/woocommerce-payments/pull/2165/commits/0bb29bd86eeef09d654fb10c5c4d28ecdd891744 . As mentioned in Slack, in my next PR I am planning on removing this check so that Subs prices are handled within `Compatibility`.